### PR TITLE
data pack loaded registries util class

### PR DESCRIPTION
**Pull request by liach**
@modmuss50 requested at https://discordapp.com/channels/507304429255393322/521545796882006027/730484313065259078
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Review by modmuss50 — APPROVED** [review]

File: mappings/net/minecraft/util/registry/Registry.mapping [hunk 1]
```diff
@@ -150,3 +150,4 @@ CLASS net/minecraft/class_2378 net/minecraft/util/registry/Registry
 		ARG 1 value
 	METHOD method_29722 getEntries ()Ljava/util/Set;
 	METHOD method_29723 isLoaded (Lnet/minecraft/class_5321;)Z
+	METHOD method_30517 getKey ()Lnet/minecraft/class_5321;
```
Comment (Earthcomputer) [inline mappings/net/minecraft/util/registry/Registry.mapping:153]: Thank goodness this has been added, I was having to use an accessor mixin >.>

File: mappings/net/minecraft/util/registry/DataPackLoadedRegistries.mapping [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	FIELD field_25933 BIOME Lnet/minecraft/class_2378;
+	FIELD field_25934 DEFAULT_VALUE_SUPPLIERS Ljava/util/Map;
+	FIELD field_25935 ROOT Lnet/minecraft/class_2385;
+	METHOD method_30559 init ()V
+	METHOD method_30560 set (Lnet/minecraft/class_2378;ILjava/lang/String;Ljava/lang/Object;)Ljava/lang/Object;
+		ARG 0 registry
+		ARG 1 rawId
+		ARG 2 id
+		ARG 3 object
+	METHOD method_30561 add (Lnet/minecraft/class_2378;Ljava/lang/String;Ljava/lang/Object;)Ljava/lang/Object;
```
Comment (Earthcomputer) [inline mappings/net/minecraft/util/registry/DataPackLoadedRegistries.mapping]: Maybe `register`, for consistency with `Registry`?

File: mappings/net/minecraft/util/registry/DataPackLoadedRegistries.mapping [hunk 3]
```diff
@@ -0,0 +1,37 @@
+CLASS net/minecraft/class_5458 net/minecraft/util/registry/DataPackLoadedRegistries
```
Comment (Earthcomputer) [inline mappings/net/minecraft/util/registry/DataPackLoadedRegistries.mapping]: This class name is a little verbose, maybe `DataPackRegistries`, or maybe `DynamicRegistries` with a javadoc comment saying what it is?

File: mappings/net/minecraft/util/registry/Registry.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/util/registry/Registry.mapping:153]: proguardthonk

File: mappings/net/minecraft/util/registry/DataPackLoadedRegistries.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/util/registry/DataPackLoadedRegistries.mapping]: Hmm, this calls `MutableRegistry#add`, hence I named that. Wonder why that is mutable registry add in the first place

File: mappings/net/minecraft/util/registry/DataPackLoadedRegistries.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/util/registry/DataPackLoadedRegistries.mapping]: Yes, `DynamicRegistries` will change

**Review by YanisBft — APPROVED** [review]

**Comment by YanisBft** [comment]
I noticed wrong names but they will be fixed with my own pr

**Review by shartte — APPROVED** [review]

**Pull request merged**